yang/vyatta-protocols-frr-isis-v1.yang

```diff
@@ -0,0 +1,822 @@
+    description "This type defines the IS-IS LSP ID format using a
+    pattern, An example LSP ID is 0143.0438.AeF0.02-01";
+  }
+
+  typedef snpa {
+    type string {
+      length "0 .. 20";
+    }
+    configd:help "This type defines the Subnetwork Point of Attachment (SNPA) format.
+    The SNPA should be encoded according to the rules specified for the particular type
+    of subnetwork being used. As an example, for an ethernet subnetwork, the SNPA is 
+    encoded as a MAC address like '00aa.bbcc.ddee'.";
+
+    description"This type defines the Subnetwork Point of Attachment (SNPA) format.
+    The SNPA should be encoded according to the rules specified for the particular type
+    of subnetwork being used. As an example, for an ethernet subnetwork, the SNPA is
+    encoded as a MAC address like '00aa.bbcc.ddee'.";
+  }
+
+  typedef system-id {
+    type string {
+      pattern "[0-9A-Fa-f]{4}\\.[0-9A-Fa-f]{4}\\.[0-9A-Fa-f]{4}";
+    }
+    configd:help "This type defines IS-IS system-id using a pattern,
+    An example system-id is 0143.0438.AeF0";
+    description "This type defines IS-IS system-id using a pattern,
+    An example system-id is 0143.0438.AeF0";
+  }
+
+  typedef net-address {
+    type string {
+      pattern "[a-fA-F0-9]{2}(\\.[a-fA-F0-9]{4}){3,9}\\.[a-fA-F0-9]{2}";
+    }
+    configd:help "This type defines an OSI NET address using a pattern,
+    An example net-address is 49.0123.6452.1972.00";
+    description "This type defines an OSI NET address using a pattern,
+    An example net-address is 49.0123.6452.1972.00";
+  }
+
+  typedef if-state-type {
+    type enumeration {
+      enum "up" {
+        value 0;
+        configd:help "Up state."; 
+        description "Up state.";
+      }
+      enum "down" {
+        value 1;
+        configd:help "Down state";
+        description "Down state";
+      }
+    }
+    configd:help "This type defines the state of an interface";
+    description "This type defines the state of an interface";
+  }
+
+  typedef adj-state-type {
+    type enumeration {
+      enum "up" {
+        value 0;
+        configd:help "State indicates the adjacency is established.";
+        description "State indicates the adjacency is established.";
+      }
+      enum "down" {
+        value 1;
+        configd:help "State indicates the adjacency is NOT established.";
+        description "State indicates the adjacency is NOT established.";
+      }
+      enum "init" {
+        value 2;
+        configd:help "State indicates the adjacency is establishing.";
+        description "State indicates the adjacency is establishing.";
+      }
+      enum "failed" {
+        value 3;
+        configd:help "State indicates the adjacency is failed.";
+        description "State indicates the adjacency is failed.";
+      }
+    }
+    configd:help "This type defines states of an adjacency";
+    description "This type defines states of an adjacency";
+  }
```

Comment:
Looks like the adj-state-type and if-state-type types are also currently unused.

Reply:
This is Resolved.